src/index.ts

```diff
@@ -65,98 +79,50 @@ export function composeStory<GenericArgs>(
   meta: Meta<GenericArgs | any>,
   globalConfig: GlobalConfig = globalStorybookConfig
 ) {
-
-  if (isInvalidStory(story)) {
-    throw new Error(
-      `Cannot compose story due to invalid format. @storybook/testing-react expected a function/object but received ${typeof story} instead.`
-    );
-  }
-
-  if (story.story !== undefined) {
-  throw new Error(
-      `StoryFn.story object-style annotation is not supported. @storybook/testing-react expects hoisted CSF stories.
-       https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#hoisted-csf-annotations`
-    );
+  if(story === undefined) {
+    throw new Error('Expected a story but received undefined.')
   }
 
-  const renderFn = typeof story === 'function' ?  story : story.render ?? globalRender;
-  const finalStoryFn = (context: StoryContext<ReactFramework, GenericArgs>) => {
-    const { passArgsFirst = true } = context.parameters;
-    if (!passArgsFirst) {
-      throw new Error(
-        'composeStory does not support legacy style stories (with passArgsFirst = false).'
-      );
-    }
-
-    return renderFn(context.args, context);
-  };
-
-  const combinedDecorators = [
-    ...(story.decorators || []),
-    ...(meta?.decorators || []),
-    ...(globalConfig.decorators || []),
-  ];
-
-  const decorated = defaultDecorateStory<ReactFramework>(
-    finalStoryFn as any,
-    combinedDecorators as any
+  const projectAnnotations = { ...defaultGlobalConfig, ...globalConfig }
```

Comment:
PS we have `composeConfig` that does a little bit more than just `Object.assign` if that feels appropriate.

Reply:
Thanks! replaced the logic with `composeConfigs`